manifests/resource/vhost.pp

```diff
@@ -616,7 +616,21 @@
     require => File[$vhost_dir],
   }
 
-  $ssl_only = ($ssl == true) and (($ssl_port + 0) == ($listen_port + 0))
+  # This deals with a situation where the listen directive for SSL doesn't match
+  # the port we want to force the SSL redirect to.
+  if ($ssl_redirect_port) {
+    $ssl_redirect_port_real = $ssl_redirect_port
```

Comment:
Current practice is to use `$_ssl_redirect_port` instead of `$ssl_redirect_port_real`